lib/vmpooler/metrics/promstats.rb

```diff
@@ -315,9 +295,9 @@ def find_metric(label)
           metric_subkey = sublabels.shift.to_sym
           raise("Invalid Metric #{metric_key}_#{metric_subkey} for #{label}") unless metric[:metric_suffixes].key? metric_subkey.to_sym
 
-          metric[:metric_name] = "#{metric[:prom_metric_prefix]}_#{metric_subkey}"
+          metric[:metric_name] = "#{@prometheus_prefix}_#{metric[:prom_metric_id]}_#{metric_subkey}"
         else
-          metric[:metric_name] = metric[:prom_metric_prefix]
+          metric[:metric_name] = "#{@prometheus_prefix}_#{metric[:prom_metric_id]}"
```

Comment:
It looks like you intend to replace `prom_metric_prefix` with `prom_metric_id` but all but one entry in the table above simply removes `prom_metric_prefix` instead.

Reply:
Have simplified it all a bit.